examples/get-insights-data/main.py

```diff
@@ -0,0 +1,121 @@
+import os
+import csv
+from utils.utils import PublicAPIClient, get_commandline_arguments
+
+
+def main():
+    args = get_commandline_arguments(
+        [
+            (
+                ["course_id"],
+                {"type": int, "help": "id of the course to get insights for"},
+            )
+        ]
+    )
+    public_api_client = PublicAPIClient(args.subdomain)
+    perspectives = fetch_course_perspectives(public_api_client, args.course_id)
+    course_data = fetch_course_data(public_api_client, args.course_id)
+
+    construct_summary(public_api_client, perspectives, course_data)
+    construct_student_insights(public_api_client, perspectives, course_data)
+
+
+def fetch_course_perspectives(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}/perspectives")
+    return response.json()["perspectives"]
+
+
+def fetch_course_data(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}")
+    return response.json()["course"]
+
+
+def construct_summary(public_api_client, perspectives, course):
+    headers = [
+        "Course ID",
+        "Course Title",
+        "Perspective UUID",
+        "Perspective Title",
+        "Views",
+        "Time Viewed [min]",
+        "Unique Viewers",
+    ]
+    csv_data = []
+    for perspective in perspectives:
+        perspective_data = [
+            course["id"],
+            course["name"],
+            perspective["uuid"],
+            perspective["title"],
+        ]
+        summary_csv = get_csv(
```

Comment:
Line 51-61 breaks on "empty" Insights. Also, the CSV parsing is a bit unstable. How about this?

```
        summary_csv = list(
            get_csv(
                public_api_client,
                f"perspectives/{perspective['uuid']}/insights/overview",
            )
        )
        csv_data.append(
            perspective_data
            + [
                get_value_from_row(summary_csv[1], "Views", 0),
                get_value_from_row(summary_csv[2], "Time Viewed [min]", 0),
                get_value_from_row(summary_csv[3], "Unique Viewers", 0),
            ]
        )
```

with a helper function:
```
def get_value_from_row(row, name, default_value):
    if row[0] != name:
        raise Exception("Insights API has changed. Please file a support week ticket.")
    try:
        return row[1]
    except IndexError:
        return default_value
```

This way if we ever change the Insights overview API endpoint, we have a better chance of catching the lack of followup here.

Reply:
good idea, I added it